From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 31 Aug 2021 22:40:09 +0000
Subject: [PATCH] SDL-1.2.15-ignore_insane_joystick_axis

changeset:   6324:95abff7adcc2
branch:      SDL-1.2
parent:      6306:2b923729fd01
user:        Ryan C. Gordon <icculus@icculus.org>
date:        Sun Jun 03 04:49:25 2012 -0400
summary:     Linux evdev: ignore joystick axis events if they aren't in a sane range.
---
 src/joystick/linux/SDL_sysjoystick.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/joystick/linux/SDL_sysjoystick.c b/src/joystick/linux/SDL_sysjoystick.c
index ee439747..2b763139 100644
--- a/src/joystick/linux/SDL_sysjoystick.c
+++ b/src/joystick/linux/SDL_sysjoystick.c
@@ -1106,6 +1106,9 @@ static __inline__ void EV_HandleEvents(SDL_Joystick *joystick)
 				}
 				break;
 			    case EV_ABS:
+				if (code > ABS_MISC) {
+					break;
+				}
 				switch (code) {
 				    case ABS_HAT0X:
 				    case ABS_HAT0Y:
